src/server/status_server/mod.rs

```diff
@@ -641,6 +664,82 @@ where
         self.thread_pool.spawn(graceful);
     }
 
+    fn start_tls_serve<I>(&mut self, builder: HyperBuilder<I>, cert_allowed_cn: HashSet<String>)
+    where
+        I: Accept<Conn = SslStream<AddrStream>, Error = std::io::Error> + Send + 'static,
+        I::Error: Into<Box<dyn StdError + Send + Sync>>,
+        I::Conn: AsyncRead + AsyncWrite + Unpin + Send + 'static,
+    {
+        let cfg_controller = self.cfg_controller.clone();
+        let router = self.router.clone();
+        // Start to tls serve.
+        let server = builder.serve(make_service_fn(move |conn: &SslStream<AddrStream>| {
+            let cfg_controller = cfg_controller.clone();
+            let router = router.clone();
+            let x509 = conn.ssl().peer_certificate();
+            let cert_allowed_cn = cert_allowed_cn.clone();
```

Comment:
It seems that they are cloned every time a new request comes. Maybe `Arc` is a better idea?

Reply:
Fixed